dashboard/src/app/dialog/dialog.component.ts

```diff
@@ -24,8 +24,7 @@ export class DialogComponent implements OnInit {
   }
 
   generatePassword() {
-    this.password = 'asdf';
-    // this.generatedPassword = passwordGenerator.genPW(); TODO: fix bug in pwgen.js, see console for error message
+    this.password = passwordGenerator.genPW();
```

Comment:
Should we check whether the password is really secure? (16 chars, lower, upper, number, special chars)

Reply:
have a look at [this](https://github.com/keycloud/keycloud/blob/master/dashboard/src/app/util/pwgen.js). You can find the generation there.
  The generated password is checked against the [haveibeenpawned](https://haveibeenpwned.com/)-api.

Reply:
[This line and the two following](https://github.com/keycloud/keycloud/blob/50602d2dc11b79a3d1b05b71c64473c645681bc6/dashboard/src/app/util/pwgen.js#L31) specify which chars are used during password generation.

Reply:
good one.